### PR TITLE
Add checking for update function

### PR DESCRIPTION
**Pull request by aguilaair**
Still needs to be added to Settings but will be done in a separate PR

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -23,6 +24,7 @@ class FvmApp extends StatelessWidget {
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
+    checkForUpdates();
```
Comment (leoafarias) [inline lib/main.dart:27]: Could we move this to a `useEffect` hook on the appShell?

File: lib/utils/manage_updates.dart [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+import '../components/molecules/update_card.dart';
+import '../version.dart';
+
+void downloadRelease(String release) async {
+  var directory = await provider.getDownloadsDirectory();
+  var fileLocation =
+      "${directory.absolute.path}${Platform.pathSeparator}CRel-$release.";
+  var url =
+      "https://github.com/aguilaair/Companion/releases/download/$release/${Platform.operatingSystem.toLowerCase()}-$release.zip";
```
Comment (leoafarias) [inline lib/utils/manage_updates.dart:21]: Need to update this for the URL on this repo. Also can we move this into the constants file?

File: lib/utils/manage_updates.dart [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+void installWindows(File file, String path) async {
+  var decompressed = ZipDecoder().decodeBytes(file.readAsBytesSync());
+  var msix =
+      decompressed.files.firstWhere((element) => element.name.contains("msix"));
+  await File("${path}msix").writeAsBytes(msix.content);
+  openLink("file://${path.replaceAll("\\", "/")}msix");
+}
+
+void installMacOS(File file, String path) async {
```
Comment (leoafarias) [inline lib/utils/manage_updates.dart:53]: I think we should download just the DMG if its possible for mac, so we would not need to decode the zip

**Review by leoafarias — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 1 above)
Comment (aguilaair) [reply lib/main.dart:27]: If I do that then every time the appShell rebuilds the popup will re-appear even if you've dismissed it.

File: lib/utils/manage_updates.dart (hunk 2 above)
Comment (aguilaair) [reply lib/utils/manage_updates.dart]: Done!

**Pull request merged**